libs/langchain-google-common/src/connection.ts

```diff
@@ -178,6 +179,22 @@ export abstract class GoogleHostConnection<
     return "gcp";
   }
 
+  get location(): string {
+    return this._location ?? this.computedLocation;
+  }
+
+  get computedLocation(): string {
```

Comment:
Any point to having a base implementation here? One less level of indirection to just put it in `location` above

Reply:
Mostly this is how location and endpoint evolved and come from a few requirements:
* The endpoint is usually, but not always, location based. So it may need to be specified separately from location.
* Other elements of the path may have location. So location may need to be specified separately from endpoint.
* In the simple cases, I didn't want people to have to specify both, but usually just location. But have them be able to if necessary.

Most resources are available at us-central1. But, apparently not all. So I wanted subclasses to be able to set their default location if appropriate.

And I wanted to be able to support default locations. Requiring a developer to have to know more magic values makes for less manageable code. Especially if we can avoid it. (We support default values all over the place.)